File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left =0;
        int right = heights.size()-1;

        int maxArea =0;

        while(left < right)
        {
            int h = min(heights[left] , heights[right]);
            int w = right -left ;
            int area = h * w;
            maxArea = max(maxArea , area);

            if(heights[left]>heights[right])
            {
                right--;
            }else {
                left ++;
            }
        }

        return maxArea;

    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.go
func max(x ,y int) int{
	
	if x>y {
		return x
	}

	return y
}

func lengthOfLongestSubstring(s string) int {
	left :=0
	longestLen :=0

	freqMap := make(map[byte]int)

	for right :=0; right<len(s); right++{
		freqMap[s[right]]++

		for freqMap[s[right]]>1{
			freqMap[s[left]]--
			left++
		}

		longestLen = max(longestLen , right-left+1)
	
	}

	return longestLen
}
